src/lib/mina_transition/validation_types.ml

```diff
@@ -0,0 +1,136 @@
+open Mina_base
+
+type ( 'time_received
+     , 'genesis_state
+     , 'proof
+     , 'delta_block_chain
+     , 'frontier_dependencies
+     , 'staged_ledger_diff
+     , 'protocol_versions )
+     t =
+  'time_received
+  * 'genesis_state
+  * 'proof
+  * 'delta_block_chain
+  * 'frontier_dependencies
+  * 'staged_ledger_diff
+  * 'protocol_versions
+
+(* TODO commented out because of weird type errors *)
```

Comment:
After discussing with you are reviewing, I think it is fine to just delete these constraints altogether.